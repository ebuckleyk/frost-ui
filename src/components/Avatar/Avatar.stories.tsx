import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { Avatar } from './Avatar';

type ComponentType = React.ComponentProps<typeof Avatar>;
const meta: Meta<ComponentType> = {
  component: Avatar,
};

export default meta;

type Story = StoryObj<ComponentType>;
export const Default: Story = {};
