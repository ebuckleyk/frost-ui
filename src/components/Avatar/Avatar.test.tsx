import * as React from 'react';
import { render } from '@testing-library/react';

import { Avatar, AvatarFallback, AvatarImage } from './Avatar';

const Component = () => {
  return (
    <Avatar>
      <AvatarImage src={'frost_ui.png'} alt="@ebuckleyk" />
      <AvatarFallback>CN</AvatarFallback>
    </Avatar>
  );
};

describe('Avatar', () => {
  it('should render Avatar component to match snapshot', () => {
    const result = render(<Component />);
    expect(result).toMatchSnapshot();
  });
});
