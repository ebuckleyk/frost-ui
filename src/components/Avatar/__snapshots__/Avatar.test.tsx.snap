// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > should render Avatar component to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
      >
        <span
          class="flex h-full w-full items-center justify-center rounded-full bg-muted"
        >
          CN
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
    >
      <span
        class="flex h-full w-full items-center justify-center rounded-full bg-muted"
      >
        CN
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
